projects/PeerTube/test.nix

```diff
@@ -0,0 +1,172 @@
+{
+  sources,
+  pkgs,
+  lib,
+  ...
+}: let
+  storageBase = "/var/peertube";
+  storageDir = subdir: "${storageBase}/${subdir}/";
+  peerUser = "peertube";
+  peerGroup = "peertube";
+  localUser = "alice";
+  localDomain = "localhost";
+  listenWeb = 9000;
+  pluginPkgs = with pkgs; [
+    peertube-plugin-akismet
+    peertube-plugin-auth-ldap
+    peertube-plugin-auth-openid-connect
+    peertube-plugin-auth-saml2
+    peertube-plugin-auto-block-videos
+    peertube-plugin-auto-mute
+    peertube-plugin-hello-world
+    peertube-plugin-logo-framasoft
+    peertube-plugin-matomo
+    peertube-plugin-privacy-remover
+    peertube-plugin-transcoding-custom-quality
+    peertube-plugin-transcoding-profile-debug
+    peertube-plugin-video-annotation
+    peertube-theme-background-red
+    peertube-theme-dark
+    peertube-theme-framasoft
+
+    peertube-plugin-livechat
+  ];
+in {
+  name = "peertube-plugins";
+
+  nodes = {
+    server = {config, ...}: {
+      imports = [
+        sources.modules.default
+        sources.modules."services.peertube.plugins"
+      ];
+
+      # Interactive testing can run into OOM with default
+      virtualisation.memorySize = 2047;
+
+      users.users."${localUser}" = {
+        description = localUser;
+        password = "foobar";
+        isNormalUser = true;
+        extraGroups = ["wheel"];
+        uid = 1000;
+      };
+
+      services.xserver = {
+        enable = true;
+        displayManager.lightdm.enable = true;
+        windowManager.icewm.enable = true;
+      };
+
+      services.displayManager = {
+        defaultSession = "none+icewm";
+        autoLogin = {
+          enable = true;
+          user = localUser;
+        };
+      };
```

Comment:
Is this needed for automated tests or just for logging into the test VM? If it's the second, one could add that in the VM test helpers: https://github.com/ngi-nix/ngipkgs/blob/main/projects/default.nix#L41-L52